ICE in get_type, at rust/ast/rust-item.h:2673

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/

-->

### Code

```rust
static STATIC_1: = 1;
```

### Version information

```
bc2a3e6
```

### Command

```
./gcc/crab1 ../test.rs -frust-debug -frust-dump-ast-pretty -Warray-bounds -dumpbase ../test.rs -mtune=generic -march=x86-64 -O0 -version -fdump-tree-gimple -o test.s -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -frust-incomplete-and-experimental-compiler-do-not-use
```

### Program output
```
../test.rs:1:18: error: unrecognised token ‘=’ in type
    1 | static STATIC_1: = 1;
      |                  ^
crab1: note: inferred crate name: test
crab1: internal compiler error: in get_type, at rust/ast/rust-item.h:2673
0x29abe86 internal_error(char const*, ...)
        ../../gccrs/gcc/diagnostic-global-context.cc:517
0xb48036 fancy_abort(char const*, int, char const*)
        ../../gccrs/gcc/diagnostic.cc:1803
0x90fe77 Rust::AST::StaticItem::get_type()
        ../../gccrs/gcc/rust/ast/rust-item.h:2673
0x90fe77 Rust::AST::StaticItem::get_type()
        ../../gccrs/gcc/rust/ast/rust-item.h:2671
0x90fe77 Rust::AST::TokenCollector::visit(Rust::AST::StaticItem&)
        ../../gccrs/gcc/rust/ast/rust-ast-collector.cc:2112
0xc62959 void Rust::AST::TokenCollector::visit<Rust::AST::Item>(std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> >&)
        ../../gccrs/gcc/rust/ast/rust-ast-collector.h:192
0xc62959 void Rust::AST::TokenCollector::visit_as_line<std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> > >(std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> >&, std::vector<std::shared_ptr<Rust::Token>, std::allocator<std::shared_ptr<Rust::Token> > >)
        ../../gccrs/gcc/rust/ast/rust-ast-collector.h:121
0xc62959 void Rust::AST::TokenCollector::visit_items_as_lines<std::vector<std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> >, std::allocator<std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> > > > >(std::vector<std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> >, std::allocator<std::unique_ptr<Rust::AST::Item, std::default_delete<Rust::AST::Item> > > >&, std::vector<std::shared_ptr<Rust::Token>, std::allocator<std::shared_ptr<Rust::Token> > >)
        ../../gccrs/gcc/rust/ast/rust-ast-collector.h:136
0xc62959 Rust::AST::TokenCollector::visit(Rust::AST::Crate&)
        ../../gccrs/gcc/rust/ast/rust-ast-collector.cc:57
0xc50556 void Rust::AST::Dump::process<Rust::AST::Crate>(Rust::AST::Crate&)
        ../../gccrs/gcc/rust/ast/rust-ast-dump.h:45
0xc93824 Rust::Session::dump_ast_pretty(Rust::AST::Crate&, bool) const
        ../../gccrs/gcc/rust/rust-session-manager.cc:1026
0xca1cfb Rust::Session::compile_crate(char const*)
        ../../gccrs/gcc/rust/rust-session-manager.cc:555
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
```
  
  </p>
</details>

--- a/gcc/rust/rust-session-manager.cc
+++ b/gcc/rust/rust-session-manager.cc
@@ -548,10 +548,6 @@ Session::compile_crate (const char *filename)
   handle_crate_name (filename, *ast_crate.get ());
 
   // dump options except lexer dump
-  if (options.dump_option_enabled (CompileOptions::AST_DUMP_PRETTY))
-    {
-      dump_ast_pretty (*ast_crate.get ());
-    }
   if (options.dump_option_enabled (CompileOptions::TARGET_OPTION_DUMP))
     {
       options.target_data.dump_target_options ();
@@ -559,6 +555,11 @@ Session::compile_crate (const char *filename)
 
   if (saw_errors ())
     return;
+
+  if (options.dump_option_enabled (CompileOptions::AST_DUMP_PRETTY))
+    {
+      dump_ast_pretty (*ast_crate.get ());
+    }
 
   // setup the mappings for this AST
   CrateNum current_crate = mappings.get_current_crate ();

